### PR TITLE
created Notification model and designed relationship between Notifica…

### PR DESCRIPTION
**Pull request by FaxriddinMaxmadiyorov**
…tion and User models

@XaoGao , task is completed, could you please review the commit?

### CODE REVIEW

File: app/models/notification.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Notification < ApplicationRecord
```
Comment (XaoGao) [inline app/models/notification.rb:1]: @FaxriddinMaxmadiyorov launch gem annotate

File: app/models/notification.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  belongs_to :sender, class_name: 'User'
```
Comment (XaoGao) [inline app/models/notification.rb:2]: @FaxriddinMaxmadiyorov before push please run rubocop and rspec

File: app/models/notification.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  belongs_to :recipient, class_name: 'User'
```
Comment (XaoGao) [inline app/models/notification.rb:3]: @FaxriddinMaxmadiyorov double quote

File: app/models/user.rb [hunk 4]
```diff
@@ -19,6 +19,7 @@ class User < ApplicationRecord
 
   has_many :vacancies, dependent: :destroy
   has_many :contacts, dependent: :destroy
+  has_many :notifications
```
Comment (XaoGao) [inline app/models/user.rb:22]: Did you test this? Witch notification will be? Recipient or sender?

File: spec/factories/notifications.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :notification do
+    sender { association(:user) }
+    recipient { association(:user) }
+    body { "MyText" }
+    read_at { "2023-09-24 11:25:45" }
```
Comment (XaoGao) [inline spec/factories/notifications.rb:6]: by defalt nil

File: spec/factories/notifications.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+FactoryBot.define do
+  factory :notification do
+    sender { association(:user) }
```
Comment (XaoGao) [inline spec/factories/notifications.rb:3]: association :sender, factory: :user
association :recipient, factory: :user

**Review by XaoGao — COMMENTED** [review]
@FaxriddinMaxmadiyorov pls fix

File: app/models/notification.rb (hunk 1 above)
Comment (FaxriddinMaxmadiyorov) [reply app/models/notification.rb:1]: @XaoGao I did it

File: app/models/notification.rb (hunk 2 above)
Comment (FaxriddinMaxmadiyorov) [reply app/models/notification.rb:2]: @XaoGao ok

File: app/models/notification.rb (hunk 3 above)
Comment (FaxriddinMaxmadiyorov) [reply app/models/notification.rb:3]: @XaoGao why we should use double quotes? as far as i know, double quotes are used while interpolating.

File: app/models/user.rb (hunk 4 above)
Comment (FaxriddinMaxmadiyorov) [reply app/models/user.rb:22]: @XaoGao you are right, I should have clarified which notifications (sent or received)

File: spec/factories/notifications.rb (hunk 6 above)
Comment (FaxriddinMaxmadiyorov) [reply spec/factories/notifications.rb:3]: @XaoGao both gives the same result, am I right?

**Comment by FaxriddinMaxmadiyorov** [comment]
@XaoGao Hey, I pushed back, could you please review the changes? Everytime I push, some tests are running, could you please explain what they are?

File: spec/models/notification_spec.rb [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+#
+#  id           :integer          not null, primary key
+#  sender_id    :integer          not null
+#  recipient_id :integer          not null
+#  body         :text
+#  read_at      :datetime
+#  created_at   :datetime         not null
+#  updated_at   :datetime         not null
+#
+require 'rails_helper'
```
Comment (XaoGao) [inline spec/models/notification_spec.rb:13]: @FaxriddinMaxmadiyorov again double quote

**Review by XaoGao — CHANGES_REQUESTED** [review]
@FaxriddinMaxmadiyorov a last moment

**Comment by FaxriddinMaxmadiyorov** [comment]
@XaoGao fixed, I asked some questions above. Could you please answer?

**Review by XaoGao — APPROVED** [review]
@FaxriddinMaxmadiyorov approve

**Pull request merged**